File: CS samples/helloworld.cs
/*! 
    \copyright Sertainty Corporation, 2020.  All Rights Reserved.

    \file     helloworld.cs

    \brief    This sample demonstrates the initialize library call
              necessary before any UXP Identity or UXP Object functions can be called.

    \author   Quinn Carter
    \date     06/17/2020

    \note     Application expects to find the necessary source files in the current working folder.
 */

using System;
using Sertainty;

namespace Sertainty
{
  using System.IO;
  

  class Helloworld
  {
    static void Main(string[] args)
    {
      Console.WriteLine("Starting Sample");

      /* Allocate a new buffer. Automatically zeros it. */

      ByteArray buffer = new ByteArray();

      /* Set up log file and initialize library.  Must do this before any active calls. */

      long ret = Sys.InitializeLibrary(buffer, args.LongLength, args, "sertainty.lic", "SertintyONE", "Sample C#", "Sample C# 1.0");
      
      if (ret == 0)
      {
        Console.WriteLine("Error initializing environment: {0}", buffer);
        Console.WriteLine("Sample failed. Ensure you have a valid installation and license.");
      }
      else
      {
        Console.WriteLine("\nHello, Sertainty!!");
        Console.WriteLine("\nYou have a valid license. Try out other samples.");
      }
    }
  }
}

File: CPP samples/helloworld.cpp
/*!
    \copyright Sertainty Corporation, 2020.  All Rights Reserved.

    \file   helloworld.cpp

    \brief  This sample demonstrates the initialize library call necessary before
            any UXP  Identity or UXP Object functions can be called.

    \author Quinn Carter
    \date   06/16/2020

    \note   Application expects to find the necessary source files in the current working folder.
 */

 #include "uxlbytearray.h"
 #include "uxpsys.h"
 #include "uxlmessage.h"

 #include <iostream>

/**
 * @brief Main entry point
 * @param argc Number of arguments
 * @param argv Array of argument tokens
 * @return 0 - success, 1 - error
 */
 int main(int argc, char *argv[])
 {
  bool licensed;

  std::cout << std::endl << "Starting Sample" << std::endl;

  /* 
      Set up log file and initialize library.
      Must do this before any active calls.
  */

  try
  {
    uxp::sys::initializeLibrary(argc, argv,"sertainty.lic","SertintyONE");
    licensed = true;
  }
  catch (uxl::exception &)
  {
    licensed = false;
  }

  uxp::sys::setLogging(uxpLogInfo);

  if (!licensed)
  {
    std::cout << "Error Error initializing environment" << std::endl;
    std::cout << "Sample failed. Ensure you have a valid installation and license." << std::endl;

    return 1;
  }

  std::cout << std::endl << "Hello, Sertainty!!" << std::endl << std::endl <<"You have a valid license. Try out other samples." << std::endl;

  return 0;
 }
